src/wp-admin/includes/upgrade.php

```diff
@@ -2290,6 +2295,23 @@ function upgrade_600() {
 	}
 }
 
+/**
+ * Executes changes made in WordPress 6.3.0
+ *
+ * @ignore
+ * @since 6.3.0
+ *
+ * @global int $wp_current_db_version The old (current) database version.
+ */
+function upgrade_630() {
+	global $wp_current_db_version;
+
+	// @TODO: Change the number to something that makes sense. This is random for the sake of this POC.
+	if ( $wp_current_db_version < 54248 ) {
+		update_option( 'wp_media_use_attachment_pages', '1' );
+	}
+}
```

Comment:
Don't think it is a good idea to store an option in the DB. This should probably be a filter. Also seems that the front-end visibility of attachment post templates should be by template type, and best would be to be controlled by the theme.